examples/gno.land/p/samourai/basedao/basedao.gno

```diff
@@ -0,0 +1,167 @@
+package basedao
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/mux"
+	"gno.land/p/samourai/daocond"
+	"gno.land/p/samourai/daokit"
+)
+
+type daoPublic struct {
+	impl *DAOPrivate
+}
+
+func (d *daoPublic) Propose(req daokit.ProposalRequest) uint64 {
+	return d.impl.Propose(req)
+}
+
+func (d *daoPublic) Execute(id uint64) {
+	d.impl.Execute(id)
+}
+
+func (d *daoPublic) Vote(id uint64, vote daocond.Vote) {
+	d.impl.Vote(id, vote)
+}
+
+// DAO is meant for internal realm usage and should not be exposed
+type DAOPrivate struct {
+	Core             *daokit.Core
+	Members          *MembersStore
+	RenderRouter     *mux.Router
+	GetProfileString ProfileStringGetter
+	Realm            std.Realm
+}
+
+type Config struct {
+	Name               string
+	Description        string
+	ImageURI           string
+	Members            *MembersStore
+	NoDefaultHandlers  bool
+	NoDefaultRendering bool
+	InitialCondition   daocond.Condition
+	SetProfileString   ProfileStringSetter
+	GetProfileString   ProfileStringGetter
+	NoCreationEvent    bool
+}
+
+type ProfileStringSetter func(field string, value string) bool
+type ProfileStringGetter func(addr std.Address, field string, def string) string
+
+const EventBaseDAOCreated = "BaseDAOCreated"
+
+func emitBaseDAOCreated() {
```

Comment:
is there a reason to have an unexported function to emit an event of creation ?

Reply:
since it's only used once, not really

Reply:
[99c4c71](https://github.com/samouraiworld/gno/pull/3/commits/99c4c7164cccac59311ec0cec43040d191d6e390)